### ISSUE TITLE
 tf.placeholder() is not compatible with eager execution.

### ISSUE THREAD

**Issue by Zuntara** [SciSharp/Keras.NET#8]
I'm running into an issue while running the basic example XOR.

I've installed python 3.6 on my system and made sure the environment variables are set correctly (it finds pyhton etc)

the problem is in the model.add(layer) function (in c# unit test):
    

        [Test]
        public void TestXor()
        {
            XOR.Run();
        }

        public class XOR
        {
            public static void Run()
            {
                // Load train data
                NDarray x = np.array(new float[,]
                                     {
                                         { 0, 0 },
                                         { 0, 1 },
                                         { 1, 0 },
                                         { 1, 1 }
                                     });
                NDarray y = np.array(new float[] { 0, 1, 1, 0 });

                // Build sequential model
                var model = new Sequential();

                var dense1 = new Dense(32, activation: "relu", input_shape: new Shape(2));
                var dense2 = new Dense(64, activation: "relu");
                var dense3 = new Dense(1, activation: "sigmoid");

                model.Add(dense1);
                model.Add(dense2);
                model.Add(dense3);

                // Compile and train
                model.Compile(optimizer: "sgd", loss: "binary_crossentropy", metrics: new string[] { "accuracy" });

                model.Summary();

                model.Fit(x, y, batch_size: 2, epochs: 1000, verbose: 1);

                // Save model and weights
                string json = model.ToJson();
                /*File.WriteAllText("model.json", json);
                model.SaveWeight("model.h5");

                // Load model and weight
                var loaded_model = Sequential.ModelFromJson(File.ReadAllText("model.json"));
                loaded_model.LoadWeight("model.h5");*/
            }
        }

produces error:

```
Python.Runtime.PythonException : RuntimeError : tf.placeholder() is not compatible with eager execution.
['  File "c:\\Users\\xxx\\AppData\\Local\\Programs\\Python\\Python36\\lib\\site-packages\\keras\\engine\\sequential.py", line 161, in add\n    name=layer.name + \'_input\')\n', '  File "c:\\Users\\xxx\\AppData\\Local\\Programs\\Python\\Python36\\lib\\site-packages\\keras\\engine\\input_layer.py", line 178, in Input\n    input_tensor=tensor)\n', '  File "c:\\Users\\xxx\\AppData\\Local\\Programs\\Python\\Python36\\lib\\site-packages\\keras\\legacy\\interfaces.py", line 91, in wrapper\n    return func(*args, **kwargs)\n', '  File "c:\\Users\\xxx\\AppData\\Local\\Programs\\Python\\Python36\\lib\\site-packages\\keras\\engine\\input_layer.py", line 87, in __init__\n    name=self.name)\n', '  File "c:\\Users\\xxx\\AppData\\Local\\Programs\\Python\\Python36\\lib\\site-packages\\keras\\backend\\tensorflow_backend.py", line 517, in placeholder\n    x = tf.placeholder(dtype, shape=shape, name=name)\n', '  File "c:\\Users\\xxx\\AppData\\Local\\Programs\\Python\\Python36\\lib\\site-packages\\tensorflow\\python\\ops\\array_ops.py", line 2140, in placeholder\n    raise RuntimeError("tf.placeholder() is not compatible with "\n']   at Python.Runtime.PyObject.Invoke(PyTuple args, PyDict kw)
   at Python.Runtime.PyObject.InvokeMethod(String name, PyTuple args, PyDict kw)
   at Python.Runtime.PyObject.TryInvokeMember(InvokeMemberBinder binder, Object[] args, Object& result)
   at CallSite.Target(Closure , CallSite , Object , PyObject )
   at System.Dynamic.UpdateDelegates.UpdateAndExecuteVoid2[T0,T1](CallSite site, T0 arg0, T1 arg1)
   at Scalper.Trader.Server.UnitTests.Engine.TensorTests.XOR.Run() 
```

My installed packages in python36:

absl-py==0.7.1
astor==0.8.0
certifi==2019.6.16
chardet==3.0.4
gast==0.2.2
google-pasta==0.1.7
grpcio==1.22.0
h5py==2.8.0
idna==2.8
joblib==0.13.2
Keras==2.2.4
Keras-Applications==1.0.8
Keras-Preprocessing==1.1.0
Markdown==3.1.1
mock==3.0.5
numpy==1.16.4
pip==19.1.1
protobuf==3.9.0
PyYAML==5.1.1
requests==2.22.0
scipy==1.3.0
setuptools==41.0.1
six==1.11.0
tensorboard==1.14.0
tensorflow==1.14.0
tensorflow-estimator==1.14.0
tensorflow-hub==0.5.0
tensorflowjs==1.2.3
termcolor==1.1.0
urllib3==1.25.3
Werkzeug==0.15.5
wheel==0.33.4
wrapt==1.11.2

any idea what's wrong?
I'm using VS2017  and .net472  installed the keras.net package

for python i installed tensorflowjs and keras   (who pulls in tensorflow 1.14)

without the tensorflowjs library i get a binding error when i call model.fit():

```
Microsoft.CSharp.RuntimeBinder.RuntimeBinderException : Cannot perform runtime binding on a null reference
   at CallSite.Target(Closure , CallSite , Object )
   at System.Dynamic.UpdateDelegates.UpdateAndExecute1[T0,TRet](CallSite site, T0 arg0)
   at Keras.Callbacks.Callback..ctor()
   at Keras.Models.BaseModel.Fit(NDarray x, NDarray y, Nullable`1 batch_size, Int32 epochs, Int32 verbose, Callback[] callbacks, Single validation_split, NDarray[] validation_data, Boolean shuffle, Dictionary`2 class_weight, NDarray sample_weight, Int32 initial_epoch, Nullable`1 steps_per_epoch, Nullable`1 validation_steps)
   at Scalper.Trader.Server.UnitTests.Engine.TensorTests.XOR.Run()
```

**Comment by deepakkumar1984** [comment]
Uninstall tensorflow and install it again using  pip

**Comment by deepakkumar1984** [comment]
Are you able to run the example?

**Comment by Zuntara** [comment]
I uninstalled tensorflow an reinstalled it (1.14), deleted everything installed again no success... also tried different versions of tensorflowjs   .... no luck.... i'm probably missing something ... not sure what...

without the tensforlowjs package I get the error:

```
Microsoft.CSharp.RuntimeBinder.RuntimeBinderException : Cannot perform runtime binding on a null reference
   at CallSite.Target(Closure , CallSite , Object )
   at System.Dynamic.UpdateDelegates.UpdateAndExecute1[T0,TRet](CallSite site, T0 arg0)
   at Keras.Callbacks.Callback..ctor()
   at Keras.Models.BaseModel.Fit(NDarray x, NDarray y, Nullable`1 batch_size, Int32 epochs, Int32 verbose, Callback[] callbacks, Single validation_split, NDarray[] validation_data, Boolean shuffle, Dictionary`2 class_weight, NDarray sample_weight, Int32 initial_epoch, Nullable`1 steps_per_epoch, Nullable`1 validation_steps)
   at Scalper.Trader.Server.UnitTests.Engine.TensorTests.XOR.Run()
```

I also removed python and reinstalled it so only the 3.6 version (latest of 3.6 is present with the environment keys set to the path (PYTHONHOME and PYTHONPATH)  in the system env vars i also reference the script, lib and main folder of python...

**Comment by Zuntara** [comment]
The runtime binding error always occurs on a "callback" object  I think it's used for some kind of logging

` at Keras.Callbacks.Callback..ctor()`

**Comment by deepakkumar1984** [comment]
Can you please share your project?

**Comment by Zuntara** [comment]
I will do my best extracting it because it's a big project :)

**Comment by Zuntara** [comment]
[XorTest.zip](https://github.com/SciSharp/Keras.NET/files/3422261/XorTest.zip)

I also left the exe in it - it also shows the crash

I found the runtime binding error - apprearantly for some reason keras.net version 0.6.4 was references instead of the latest one.  After updating that one I get the other error (eager exception)  so the callback error seems to be fixed.

the backend in keras.json  is tensorflow 

{
    "image_data_format": "channels_last",
    "floatx": "float32",
    "backend": "tensorflow"
}

**Comment by deepakkumar1984** [comment]
Can you check the numpy version, if it 3.6.1.6, can you downgrade to 3.6.1.1 and test it?

**Comment by deepakkumar1984** [comment]
The XOR test you shared worked for me, so should be something missing in your python environment
![image](https://user-images.githubusercontent.com/6328361/61918939-88fed980-af92-11e9-93c1-7bf86716ab53.png)

**Comment by deepakkumar1984** [comment]
Please reopen if still having issue
